src/cassandra.go

```diff
@@ -103,6 +101,64 @@ func main() {
 	fatalIfErr(i.Publish())
 }
 
+// runMetricCollection will perform the metrics collection.
+func runMetricCollection(i *integration.Integration, jmxClient *gojmx.Client) error {
+	if args.LongRunning {
+		return collectMetricsEachInterval(i, jmxClient)
+	}
+	return collectMetrics(i, jmxClient)
+}
+
+// collectMetricsEachInterval will collect the metrics periodically when configured in long-running mode.
+func collectMetricsEachInterval(i *integration.Integration, jmxClient *gojmx.Client) error {
+	metricInterval := time.NewTicker(time.Duration(args.Interval) * time.Second)
+
+	runHeartBeat()
+
+	// do ... while.
+	for ; true; <-metricInterval.C {
+		// Check if the nrjmx java sub-process is still alive.
+		if !jmxClient.IsRunning() {
+			return errNRJMXNotRunning
+		}
+
+		if err := collectMetrics(i, jmxClient); err != nil {
+			log.Error("Failed to collect metrics, error: %v", err)
+			continue
+		}
+
+		if err := i.Publish(); err != nil {
+			log.Error("Failed to publish metrics, error: %v", err)
+			continue
+		}
+	}
+
+	return nil
+}
+
+// collectMetrics will gather all the required metrics from the JMX endpoint and attach them the the sdk integration.
+func collectMetrics(i *integration.Integration, jmxClient *gojmx.Client) error {
+	e, err := entity(i)
```

Comment:
if I am not wrong this will be always the same entity even if we retrieve it each interval. Right?

```
// Entity method creates or retrieves an already created entity.

```

Reply:
Oh, no, i.Publish() should delete them :)